librepo/downloader.c

```diff
@@ -1836,6 +1836,25 @@ check_finished_transfer_status(CURLMsg *msg,
                         target->mirror->max_ranges);
             }
             return TRUE;
+        } else if (target->target->zck_dl != NULL && zck_is_error(zck_dl_get_zck(target->target->zck_dl)) > 0) {
+            zckCtx *zck = zck_dl_get_zck(target->target->zck_dl);
+
+            // Something went wrong while writing the zchunk file
+            g_set_error(transfer_err, LR_DOWNLOADER_ERROR, LRE_ZCK,
+                        "Zchunk error: %s",
+                        zck_get_error(zck));
+            if (zck_is_error(zck) == 1) {
+                // Non-fatal zchunk error
+                g_info("Serious zchunk error: %s",
+                   zck_get_error(zck));
+                *serious_error = TRUE;
+            } else { // zck_is_error(zck) == 2
+                // Fatal zchunk error
+                g_info("Fatal zchunk error: %s",
+                   zck_get_error(zck));
+                *fatal_error = TRUE;
+            }
+            return TRUE;
```

Comment:
The above change will work with any version of zchunk, but zchunk versions < 1.1.5 will give somewhat cryptic error messages.

Reply:
FWIW, I completely agree that any error messages should include the file name.  In my original version of this code, the filename was actually listed twice.  `zck_get_error()` doesn't contain the file name (since the `zck` struct doesn't store that information), but whichever of these errors gets triggered will be appended to the error message in line 1514 which also includes the filename.

Sample message:
```Error: Failed to download metadata for repo 'updates': Yum repo downloading error: Downloading error(s): repodata/8b1f8fa78ab3f87e20e0e3a9a473fc23c303278947be95065e4845da17106dd6-filelists.xml.zck - Download failed: Zchunk error: Error writing data: No space left on device```

Reply:
Oh, cool, sorry I've missed that.

Reply:
No problem at all.  It's nice to know you're looking closely enough at the PR to ask these questions.